pkg/app/piped/cloudprovider/kubernetes/loader.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipecd/pkg/app/piped/toolregistry"
+	"github.com/pipe-cd/pipecd/pkg/config"
+	"github.com/pipe-cd/pipecd/pkg/git"
+)
+
+type TemplatingMethod string
+
+const (
+	TemplatingMethodHelm      TemplatingMethod = "helm"
+	TemplatingMethodKustomize TemplatingMethod = "kustomize"
+	TemplatingMethodNone      TemplatingMethod = "none"
+)
+
+type ManifestLoader interface {
+	// LoadManifests renders and loads all manifests for application.
+	LoadManifests(ctx context.Context) ([]Manifest, error)
+}
```

Comment:
nits, I would prefer either fixing to `ManifestLoader` into `Loader` or `LoadManifests` into `Load`.

Reply:
Nice idea. Let me fix the interface name to `Loader`. 👍

Reply:
Addressed by https://github.com/pipe-cd/pipecd/pull/3407/commits/f1d6831f41d9d0ae083d649e7aa8d0ebc52962aa